vignettes/qgisprocess.Rmd

```diff
@@ -0,0 +1,271 @@
+---
+title: "Basic usage of qgisprocess"
+author: "Jannes Muenchow & Floris Vanderhaeghe"
+date: "Last updated: yyyy-mm-dd"
+output:
+  html_document: 
+    toc: yes
+    toc_float:
+      collapsed: no
+      smooth_scroll: no
+    toc_depth: 2
+vignette: >
+  %\VignetteIndexEntry{Basic usage of qgisprocess}
+  %\VignetteEncoding{UTF-8}
+  %\VignetteEngine{knitr::knitr}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+# Setting up the system
+
+**qgisprocess** is basically a wrapper around the standalone command line tool [`qgis_process`](https://github.com/qgis/QGIS-Documentation/issues/5803).
+Therefore, you need to have installed QGIS on your system as well as third-party providers such as GRASS and SAGA to access and run all geoalgorithms provided through `qgis_process` from within R.
+
+To facilitate using **qgisprocess**, we have created a docker image that already comes with the needed software packages.
+You can pull it from Github's container registry by running:
+
+```{sh, eval=FALSE}
+docker pull ghcr.io/geocompx/docker:qgis
+```
+
+For a more detailed introduction on how to get started with docker, please refer to https://github.com/geocompx/docker.
+
+# Package configuration
+
+Since **qgisprocess** depends on the command line tool `qgis_process`, it already tries to detect `qgis_process` on your system when it is being loaded, and complains if it cannot find it.
+
+```{r}
+library("qgisprocess")
+```
+
+In fact, once **qgisprocess** has detected `qgis_process`, it will cache among others the path to the executable, the QGIS version and the list of known algorithms.
```

Comment:
```suggestion
When loading **qgisprocess** for the first time, it will cache among others the path to `qgis_process`, the QGIS version and the list of known algorithms.
When loading **qgisprocess** in later R sessions, the cache file is read instead for speed-up, on condition that it is still valid.
```